### ISSUE TITLE
Make the project easier for community members to get running locally

### ISSUE THREAD

**Issue by camfindlay** [silverstripe/addons.silverstripe.org#88]
Currently the Vagrant and puppet setup is unavailable to people outside SS Ltd. Running locally without this makes it difficult and time consuming to just get set up to even attempt a contribution to the code.

Ideally a contributor should be able to clone and run a `vagrant up` to get all dependancies required to start making improvements to the code.

**Comment by dhensby** [comment]
Hi @camfindlay 

The new https://github.com/betterbrief/vagrant-skeleton does this now.

You can just download the files, install vagrant and virtual box and run `vagrant up` and it'll install a fresh SS out of the box.

I'm not sure if you use this internally at all, but I/we could make a walk through of how to get an SS machine running with vagrant and submitting a PR?

**Comment by camfindlay** [comment]
Sounds good, I actually use a fork of the vagrant-skeleton for some of our training.

Would be good to have docs to get addons running more easily for community contributors... you'd have to make sure you perhaps include an elasticsearch and redis install specifically in this case as the app relies on this things.

**Comment by chillu** [comment]
Yeah, my vagrant setup on addons always had the flaw of relying on SS internal puppet code - I was hoping that was sufficiently modularised at some point to open source, but there's no driver for it. In hindsight, I think its more important to get devs going quickly with non-trivial dependencies like Redis than to accurately simulate our live environment - so +1 on this.

Note that the current live setup doesn't use Redis (because its a pain in the ass to set up and monitor).

**Comment by camfindlay** [comment]
in that case perhaps a betterbrief vagrant skeleton setup which also includes elastic search preconfigured might help? I also created this simple vagrant setup which installs and fires up elasticsearch though I haven't had time to tidy up the provision script: https://github.com/camfindlay/addons.silverstripe.org/blob/bashvagrant/vagrant.sh

It's just running a basic trusty32 ubuntu box which seems to be pretty commonly used.

**Comment by dhensby** [comment]
> Would be good to have docs to get addons running more easily for community contributors... you'd have to make sure you perhaps include an elasticsearch and redis install specifically in this case as the app relies on this things.

ah yes, sorry, I didn't see that you were specifically referring to addons. I think that we can add a way to install elasticsearch via a provisioner script quite easily.

One approach could be to have a specific box (rather than a set of provisioners) that could be used instead?

**Comment by robbieaverill** [comment]
Hey guys - is this something that changes with recipes? There's possibly a question to be answered around which provisioning system to use for a public "package" to install. I'd personally be in favour of a docker setup - would allow users to still run containerized redis and ES instances and have a production-like test environment. Of course it wouldn't work very well for Windows users without Hyper V...
